sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -297,7 +297,8 @@ TEST(AzureCliCredential, ContextCancelled)
   TokenRequestContext trc;
   trc.Scopes.push_back("https://storage.azure.com/.default");
 
-  auto context = Context{std::chrono::system_clock::now() + std::chrono::hours(24)};
+  auto context
+      = Context::CreateWithDeadline(std::chrono::system_clock::now() + std::chrono::hours(24));
```

Comment:
The ctor overload approach worked well and made sense.
Why have we decided to go back to the factory method approach?

Imo, we should revert this last commit, as the API surface that was previously had was good.